unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/FriendsController/Tests/RPCSocialApiBridgeShould.cs

```diff
@@ -0,0 +1,124 @@
+                         {
+                             foreach (var users in usersList)
+                             {
+                                 if (token.IsCancellationRequested) break;
+                                 await writer.YieldAsync(users); // instead of `yield return`
+                             }
+                         }));
+
+            clientSocial.GetRequestEvents(new Payload() { SynapseToken = "" })
+                        .Returns(new UniTask<RequestEvents>(new RequestEvents()
+                         {
+                             Incoming = new Requests()
+                             {
+                                 Items =
+                                 {
+                                     new List<RequestResponse>()
+                                     {
+                                         new RequestResponse()
+                                         {
+                                             User = new User() { Address = "addr1" },
+                                             Message = "",
+                                             CreatedAt = 1678912696639
+                                         },
+                                         new RequestResponse()
+                                         {
+                                             User = new User() { Address = "addr2" },
+                                             Message = "",
+                                             CreatedAt = 1678912696639
+                                         }
+                                     }
+                                 }
+                             },
+                             Outgoing = new Requests()
+                             {
+                                 Items =
+                                 {
+                                     new List<RequestResponse>()
+                                     {
+                                         new RequestResponse()
+                                         {
+                                             User = new User() { Address = "addr3" },
+                                             Message = "",
+                                             CreatedAt = 1678912696639
+                                         },
+                                         new RequestResponse()
+                                         {
+                                             User = new User() { Address = "addr4" },
+                                             Message = "",
+                                             CreatedAt = 1678912696639
+                                         }
+                                     }
+                                 }
+                             }
+                         }));
+
+            UserProfile.GetOwnUserProfile().UpdateData(new UserProfileModel() { userId = OWN_ID });
```

Comment:
instead of setting the static reference, you should mock the `userProfileBridge`:
```
var ownProfile = ScriptableObject.CreateInstance<UserProfile>();
ownProfile.UpdateData(new UserProfileModel { ... });
userProfileBridge.GetOwn().Returns(ownProfile);
```